pkg/daemon/server/service/pipeline_metrics_query.go

```diff
@@ -156,61 +156,73 @@ func (ps *pipelineMetadataQuery) GetVertexMetrics(ctx context.Context, req *daem
 			Name: vertexName,
 		},
 	}
-
-	processingRates := make(map[string]float64, 0)
-	pendings := make(map[string]int64, 0)
+	podNum := int64(1)
+	// for now only reduce has parallelism might have to modify later
+	obj := ps.pipeline.GetFromEdges(req.GetVertex())
```

Comment:
I understand we do not need to check if it's reduce vertex because the parallelism is nil if it's not. Adding more comments to mention that would be appreciated - I thought a while before figuring it out. :)

Reply:
sure